### PR TITLE
Move nbt ops to nbt package

### LINKED ISSUE #1649: NbtOps$MapBuilder and CompoundTag have to be in the same package
`net/minecraft/class_2509$class_5320` (`net/minecraft/datafixer/NbtOps$MapBuilder`) and `net/minecraft/class_2487` (`net/minecraft/nbt/CompoundTag`) have to be in the same package.

In the `DataResult<Tag> build(CompoundTag compoundTag, Tag tag)` method of `NbtOps$MapBuilder`, it calls `method_29143` (unnamed, maybe `toMap`), which is a protected method in `CompoundTag`. Since `MapBuilder` is not a subclass of `CompoundTag`, it is fairly clear they should be in the same package.

I personally suggest moving NbtOps to the nbt package.

### PR DESCRIPTION
**Pull request by liach**
Fixes #1649

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by i509VCB — APPROVED** [review]

**Review by Earthcomputer — APPROVED** [review]
🤢

File: mappings/net/minecraft/nbt/NbtOps.mapping [hunk 1]
```diff
@@ -0,0 +1,21 @@
+CLASS net/minecraft/class_2509 net/minecraft/nbt/NbtOps
+	COMMENT Used to handle Minecraft NBTs within {@link com.mojang.serialization.Dynamic
+	COMMENT dynamics} for DataFixerUpper, allowing generalized serialization logic
+	COMMENT shared across different type of data structures. Use {@link NbtOps#INSTANCE}
+	COMMENT for the ops singleton.
```
Comment (shartte) [inline mappings/net/minecraft/nbt/NbtOps.mapping:5]:
```suggestion
	COMMENT shared across different types of data structures. Use {@link NbtOps#INSTANCE}.
```

**Review by shartte — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Review by Mysterious-Dev — APPROVED** [review]

File: mappings/net/minecraft/nbt/NbtOps.mapping (hunk 1 above)
Comment (Earthcomputer) [reply mappings/net/minecraft/nbt/NbtOps.mapping:5]: Kind of irrelevant to this PR, which is only a repackage

**Review by modmuss50 — APPROVED** [review]
LGTM

**Pull request merged**